packages/core/src/typings/openapi.ts

```diff
@@ -112,13 +112,10 @@ export interface Oas3Xml {
 }
 
 // common fields for OpenAPI Schema v3.x
-interface Oas3XSchemaBase<T> {
+interface Oas3XSchemaBase {
```

Comment:
I prefer using the generic interface because it's less likely for the common properties to diverge. Not sure which approach would be cleaner though. Why did you decide to change the approach?

Reply:
the keywords which utilized the generic interface were moved into the `Oas3Schema` and `Oas3_1Schema`.  So it was unused.

Reply:
Yes, but what was the reason for moving the keywords from `Oas3XSchemaBase`? Was there an issue with leaving them as they were?

Reply:
Yes, they have different behavior between 30 asd  31.  So the extension of Oas3XSchemaBase  was incorrect and the keywords should be assigned to each version

Reply:
Ok. I'll test that once I have time. Thank you for contributing!